### DISCUSSION TITLE
Add a surcharge on the total price  [HELP]

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#10738] [General] [answered]
Hi,
We have a product that can be customized with a sign.
If the customer select this option a surcharge will be applied on the total price.

It's possible to do that with saleor?

Thanks in advance and kind regards,

Davide

**Comment by DidoMarchet** [comment]
Hi,
do you have any advice ?!

Really thanks and kind regards,

Davide

**Comment by peelar** [comment, marked as answer]
Hello @DidoMarchet 👋 

What do you mean by "a sign"? Is using Variants an option for you? (e.g. variant 1 with no surcharge, variant 2 with surcharge)

**Reply by DidoMarchet** [reply]
Hi @peelar , sorry for my english !
So, we have a product (a glass) and the variant is a combination of color of the lens and size.
The variant has its own price and SKU.

However to this product we can add an option (add your name to the frame eg.) and the cost of this option is 20 euro. 
This cost is not part of the price of the variant but it's a surcharge we need to add to the total amount of the order.

Add a variant with option can be an idea but we have a great number of variant for each product (a matrix from color lens and size) and then it's difficult even because the option on the variant must not change the sku.

Thanks and kind regards,

Davide

**Reply by Anubiella** [reply]
Good morning everyone, 
always nice to be here. 👍 

Hi @peelar ,

what we're saying is: are there other options within Saleor to take into account (_and add it in the checkout price/s_) 
other surplus/extra fees without the creation of _x_ variants?
I mean, we have already a complex matrix of size/colors variants: adding other specific variant of variants will be honestly too much, 
that's why we're looking for an alternative.

E.g.
Let's say we have a product in SL with its own attributes etc , and its own variants (with their own specific attributes and SKUs),
for a unit price of $100, ok.
Let's say there are specific situation where the customer can choose a service which has its own price, say it's 20$.
Say also that this service can be cross-product, it should be available for many products.

So ok I can put a checkbox (say an attribute) within the product page in SL, but how can we handle this price changing behaviour without creating x variants? Has that situation been planned?


Thanks a lot in advance,
god bless

**Reply by IKarbowiak** [reply]
Hi,

You can override the price of the checkout line with the use of an app, but I'm not sure if it resolves your problem. You can read more about that [here](https://docs.saleor.io/docs/3.x/developer/checkout#setting-custom-prices-for-checkout-items).

**Reply by DidoMarchet** [reply]
Hi @IKarbowiak,
thanks! We'll try!
have a nice day and regards,

Davide

**Reply by Anubiella** [reply]
Hi @IKarbowiak ,

thanks for your hint, we'll try and let you know asap, it could be a resolutive intervention.

Thanks a lot,
Lorenzo

**Reply by DidoMarchet** [reply]
> Hi,
> 
> You can override the price of the checkout line with the use of an app, but I'm not sure if it resolves your problem. You can read more about that [here](https://docs.saleor.io/docs/3.x/developer/checkout#setting-custom-prices-for-checkout-items).

It works perfectly.
